tracpro/orgs_ext/tasks.py

```diff
@@ -88,19 +94,30 @@ def apply_async(self, *args, **kwargs):
         return super(OrgTask, self).apply_async(*args, **kwargs)
 
     def check_rate_limit(self, org):
-        """Return True if the task has been run too recently for this org."""
-        now = get_now()
-        last_run_key = LAST_RUN_KEY.format(task=self.__name__, org=org.pk)
+        """Return the next run time if this task has run too recently."""
+        now = timezone.now()
+        last_run_key = self.get_cache_key(org, LAST_RUN_TIME)
+        fail_count_key = self.get_cache_key(org, FAILURE_COUNT)
+
         last_run_time = cache.get(last_run_key)
         if last_run_time is not None:
+            # Calculate how long until the task is eligible to run again.
             last_run_time = parse_iso8601(last_run_time)
-            if now - last_run_time < settings.ORG_TASK_TIMEOUT:
-                return True
+            delta = settings.ORG_TASK_TIMEOUT * 2 ** cache.get(fail_count_key, 0)
+            delta = max(delta, datetime.timedelta(days=1))
+            if now - last_run_time < delta:
+                return last_run_time + delta
+
+        # Set the current time as the last run time and allow the task to run now.
         cache.set(last_run_key, format_iso8601(now))
```

Comment:
This appears to use the default cache timeout (I don't see it configured so it would be 5 mins by default) meaning this key will expire before it should be released.